src/backend/distributed/commands/extension.c

```diff
@@ -113,5 +110,661 @@ ExtractNewExtensionVersion(Node *parsetree)
 		}
 	}
 
-	return newVersion;
+	/* return target string safely */
+	if (newVersion)
+	{
+		return pstrdup(newVersion);
+	}
+	else
+	{
+		return NULL;
+	}
+}
+
+
+/*
+ * PlanCreateExtensionStmt is called during the creation of an extension.
+ * It is executed before the statement is applied locally.
+ * We decide if the extension needs to be replicated to the worker, and
+ * if that is the case return a list of DDLJob's that describe how and
+ * where the extension needs to be created.
+ */
+List *
+PlanCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const char *queryString)
+{
+	List *commands = NIL;
+	const char *createExtensionStmtSql = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return NIL;
+	}
+
+	/* extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	/*
+	 * Make sure that no new nodes are added after this point until the end of the
+	 * transaction by taking a RowShareLock on pg_dist_node, which conflicts with the
+	 * ExclusiveLock taken by master_add_node.
+	 * This guarantees that all active nodes will have the extension, because they will
+	 * either get it now, or get it in master_add_node after this transaction finishes and
+	 * the pg_dist_object record becomes visible.
+	 */
+	LockRelationOid(DistNodeRelationId(), RowShareLock);
+
+	/*
+	 * Make sure that the current transaction is already in sequential mode,
+	 * or can still safely be put in sequential mode
+	 */
+	EnsureSequentialModeForExtensionDDL();
+
+	/*
+	 * Here we append "schema" field to the "options" list (if not specified)
+	 * to satisfy the schema consistency between worker nodes and the coordinator.
+	 */
+	AddSchemaFieldIfMissing(createExtensionStmt);
+
+	createExtensionStmtSql = DeparseTreeNode((Node *) createExtensionStmt);
+
+	/*
+	 * To prevent recursive propagation in mx architecture, we disable ddl
+	 * propagation before sending the command to workers.
+	 */
+	commands = list_make3(DISABLE_DDL_PROPAGATION,
+						  (void *) createExtensionStmtSql,
+						  ENABLE_DDL_PROPAGATION);
+
+	return NodeDDLTaskList(ALL_WORKERS, commands);
+}
+
+
+/*
+ * Add DefElem item for "schema" (if not specified in statement) to "options"
```

Comment:
Minor: We typically start the function comment with the functions name  `AddSchemaFieldIfMissing  adds ...`

(please apply the  same to all functions)

Reply:
Done in 26e1acfcd1144e0ee2ed3527e51e8122ee340143